chainerx_cc/chainerx/cuda/memory_pool.cc

```diff
@@ -289,6 +302,11 @@ void MemoryPool::Free(void* ptr) {
 
         PushIntoFreeList(std::move(chunk));
     }
+
+    if (free_postprocess_hook_) {
+        // TODO(mkusumoto): Is this a valid call?
+        free_postprocess_hook_(ptr, *this);
```

Comment:
This should be a "valid" call, but I think maybe the `ptr` argument is useless, because at the moment of this call, another `Malloc` call could have assigned this address to a different allocation.

Are there any use cases for this callback? 🤔

Reply:
Honestly, I cannot think of use case for this part. It may be good to remove this function hook to keep the implementation a little simpler.

Reply:
I removed `free_postprocess_hook`.